docs/guides/dart/flutter.mdx

```diff
@@ -0,0 +1,1143 @@
+```
+
+We can then scaffold the project using the following command:
+
+```bash
+flutter create word_generator
+```
+
+Then open your project in your favorite editor:
+
+```bash
+code word_generator
+```
+
+## Backend
+
+Let's start by building out the backend. This will be an API with a route dedicated to getting a list of all the favorites and a route to toggle a favorite on or off. These favorites will be stored in a [key value store](/keyvalue). To create a Nitric project add the `nitric.yaml` to the Flutter template project.
+
+```yaml {{ label: "nitric.yaml" }}
+name: word_generator
+services:
+  - match: lib/services/*.dart
+    start: dart run $SERVICE_PATH
+```
+
+This points the project to the services that we will create in the `lib/services` directory. We'll create that directory and a file to start building our services.
+
+```bash
+mkdir lib/services
+touch lib/services/main.dart
+```
+
+You will also need to add Nitric to your `pubspec.yaml`.
+
+```bash
+flutter pub add nitric_sdk
+```
+
+### Building the API
+
+Define the API and the key value store in the `main.dart` service file. This will create an API named `main`, a key value store named `favorites`, and the function permissions to get, set, and delete documents. The `favorites` store will contain keys with the name of the favorite and then a value with the favorites object.
+
+```dart {{ label: "lib/services/main.dart" }}
+import 'package:nitric_sdk/nitric.dart';
+
+void main() {
+  final api = Nitric.api("main");
+
+  final favoritesKV = Nitric.kv("favorites").allow([
+    KeyValueStorePermission.get,
+    KeyValueStorePermission.set,
+    KeyValueStorePermission.delete
+  ]);
+}
+```
+
+We will define a favorites class to convert our JSON requests to favorite objects and then back into JSON. Conversion to a `Map<String, dynamic>` will also allow us to store the favorites objects in the key value store. We can do this by defining `fromJson` and `toJson` methods, allowing the builtin `jsonEncode` and `jsonDecode` methods to understand our model. By defining this as a class it unifies the way the frontend and backend handle favorites objects, while leaving room for extension for additional metadata.
+
+```dart {{ label: "lib/favorite.dart" }}
+class favorite {
+  /// The name of the favorite
+  String name;
+
+  favorite(this.name);
+
+  /// Convert a json decodable map to a favorite object
+  favorite.fromJson(Map<String, dynamic> json) : name = json['name'];
+
+  /// Convert a favorite object to a json encodable
+  static Map<String, dynamic> toJson(favorite favorite) =>
+    {'name': favorite.name};
+}
+```
+
+For the API we will define two routes, one GET method for `/favorites` and one POST method on `/favorite`. Let's start by defining the GET `/favorites` route. Make sure you import `dart:convert` to get access to the `jsonEncode` method for converting the documents to favorites.
+
+```dart {{ label: "lib/services/main.dart" }}
+import 'dart:convert';
+
+...
```

Comment:
These can now be replaced by the collapse functionality. Repeat for the test of the snippets